### PR TITLE
Add init_with declaration

### PR DESCRIPTION
**Pull request by palkan**

### CODE REVIEW

File: lib/clowne/adapters/sequel.rb [hunk 1]
```diff
@@ -6,17 +6,8 @@ module Clowne
...
-      # +params+:: Custom params hash
-      def clone(source, plan, params: {})
-        declarations = plan.declarations
-        init_record = RecordWrapper.new(dup_source(source))
-
-        declarations.inject(init_record) do |record, (type, declaration)|
-          resolver_for(type).call(source, record, declaration, params: params)
-        end
+      def init_record(source)
+        RecordWrapper.new(source)
```
Comment (ssnickolay) [inline lib/clowne/adapters/sequel.rb]: it's cool for me :+1:

File: lib/clowne/adapters/sequel/record_wrapper.rb [hunk 2]
```diff
@@ -6,31 +6,6 @@ class Sequel
       class RecordWrapper
         attr_reader :record, :association_store
 
-        class << self
-          def to_hash(record_wrapper)
```
Comment (ssnickolay) [inline lib/clowne/adapters/sequel/record_wrapper.rb]: Damn, I forgotten about this >_< thx :+1:

File: lib/clowne/adapters/registry.rb [hunk 3]
```diff
@@ -31,12 +31,12 @@ def insert_before(before, action)
       end
 
       def append(action)
-        validate_uniq!(action)
+        actions.delete(action)
         actions.push action
       end
 
       def prepend(action)
```
Comment (ssnickolay) [inline lib/clowne/adapters/registry.rb]: May be rename `prepend` -> `unshift`? Because `prepend` reserved word.

File: docs/init_with.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+end
+
+class UserCloner < Clowne::Cloner
+  adapter :active_record
+
+  include_association :profile
+
+  trait :copy_settings do
+    # Use a `target` for all the actions
+    init_with { |_source, target:| target }
```
Comment (ssnickolay) [inline docs/init_with.md]: What do you think about `init_as` instead of `init_with`?

File: lib/clowne/adapters/registry.rb (hunk 3 above)
Comment (ssnickolay) [reply lib/clowne/adapters/registry.rb]: p/s/ I "did't see" it before о_о

File: docs/init_with.md (hunk 4 above)
Comment (palkan) [reply docs/init_with.md]: Good idea

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 ## master
 
+- Add `init_with` declaration. ([@palkan][])
```
Comment (ssnickolay) [inline CHANGELOG.md]: `init_as` ;)

**Pull request merged**